### ISSUE TITLE
maybe_single throws an error when 0 rows are returned

### ISSUE THREAD

**Issue by levity** [supabase/supabase-py#1207]
# Bug report

## Describe the bug

[maybe_single](https://postgrest-py.readthedocs.io/en/latest/api/request_builders.html#postgrest.SyncSelectRequestBuilder.maybe_single) is supposed to return None when the query returns no rows. But now it throws an APIError:

```postgrest.exceptions.APIError: {'message': 'Missing response', 'code': '204', 'hint': 'Please check traceback of the code', 'details': "Postgrest couldn't retrieve response, please check traceback of the code. Please create an issue in `supabase-community/postgrest-py` if needed."}```

## To Reproduce

Run any query that returns 0 rows, e.g:

`postgrestClientInstance.from_('my_table').select('*').eq('content', 'nonexistent').maybe_single().execute()`

## Expected behavior

should return `APIResponse(data=None)` as in previous versions

## System information

- supabase 2.3.0
- postgrest 0.13.0
- PostgreSQL 15.1 (Ubuntu 15.1-1.pgdg20.04+1) on aarch64-unknown-linux-gnu, compiled by gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0, 64-bit

**Comment by DhenPadilla** [comment]
Wondering if there's an update on this?

**Comment by Krishna-js45** [comment]
Thanks for the clarification. Understood that this is a breaking-change candidate and will be addressed as part of v3.0. Appreciate the context.

**Comment by o-santi** [comment]
> Wondering if there's an update on this?

This will be fixed in the next major version. See https://github.com/supabase/supabase-py/pull/1370 for current status.

**Comment by gostak-dd** [comment]
tldr maybe_single() asks if there's a match, gets a weird response it doesn't understand, and silently breaks down. much like me on Hinge...

**Comment by o-santi** [comment]
I believe I found a very simple solution that is not a breaking change, in #1424. I'll merge it for now, while we don't have the next major version ready for release.
